scripts/limonade/src/index.ts

```diff
@@ -67,16 +71,49 @@ async function run() {
   let { blockhash: latestBlockhash } = await connection.getLatestBlockhash(
     "confirmed"
   );
+  const removeClosedAccounts = async () => {
+    const response = await connection.getProgramAccounts(limoId, {
+      commitment: "confirmed",
+      filters,
+      withContext: true,
+    });
+    existingAccounts.forEach(async (key) => {
+      if (response.value.find((account) => account.pubkey.toBase58() === key)) {
+        return;
+      }
+
+      try {
+        await client.removeOpportunity({
+          program: "limo",
+          chainId: argv.chainId,
+          permissionAccount: new PublicKey(key),
+          router: getPdaAuthority(limoId, globalConfig),
+        });
+        existingAccounts.delete(key);
+      } catch (e) {
+        console.error("Failed to remove opportunity", e);
+      }
+    });
+  };
   const submitExistingOpportunities = async () => {
     const response = await connection.getProgramAccounts(limoId, {
       commitment: "confirmed",
       filters,
       withContext: true,
     });
+
+    response.value.forEach((account) => {
+      existingAccounts.add(account.pubkey.toBase58());
+    });
+
     const payloads: OpportunityCreate[] = response.value
+      .filter(
+        (account) =>
+          lastChange[account.pubkey.toBase58()] === undefined ||
+          lastChange[account.pubkey.toBase58()] < Date.now() - 60 * 1000
+      )
       .map((account) => ({
         program: "limo" as const,
-        blockHash: latestBlockhash,
         chainId: argv.chainId,
         slot: response.context.slot,
         order: {
```

Comment:
nit: a few lines below: the `Initial opportunities` log makes less sense now since it's in a for loop. maybe change it to something like resubmitting opportunities?